### PR TITLE
150 warnings

### LINKED ISSUE #150: Clean up React warnings
```
"Warning: Each child in an array or iterator should have a unique "key" prop. Check the render method of `FilterBar`. See https://fb.me/react-warning-keys for more information."react-dev.js:20749:9
Warning: Failed propType: Required prop `openVariablesView` was not specified in `ConsoleOutput`. Check the render method of `Connect(ConsoleOutput)`.react-dev.js:20749:9
Warning: Failed propType: Invalid prop `messages` of type `object` supplied to `ConsoleOutput`, expected `array`. Check the render method of `Connect(ConsoleOutput)`.
```

### PR DESCRIPTION
**Pull request by linclark**
Fixes #150 
## Testing instructions

Run test in #201

### CODE REVIEW

**Comment by linclark** [comment]
@bgrins this should be ready for testing with your PR

**Comment by bgrins** [comment]
This looks good to me, thanks

File: devtools/client/webconsole/new-console-output/components/filter-bar.js [hunk 1]
```diff
@@ -127,7 +127,9 @@ const FilterBar = createClass({
 
     return (
       dom.div({className: "webconsole-filteringbar-wrapper"},
-        children
+        // The elements in FilterBar will never change order so we can use the array
```
Comment (linclark) [inline devtools/client/webconsole/new-console-output/components/filter-bar.js]: I just realized this isn't true... let me fix this before we commit, but it should be fine for testing your PR

**Comment by linclark** [comment]
The issue with what I did [here](https://github.com/devtools-html/gecko-dev/pull/204/files/868270c516886f56fdf53bef5626812b7c1594a1..30b5c267019aa9c5ed641b5dbc072d454724ca75#r73591037) was that some array items might disappear from the array based on visibility, so their array indexes (and thus their keys) would have changed. I changed this to `...children`. Doing a spread is a better solution here anyway and is what I should have done in the first place.

**Pull request merged**